vulcain.go

```diff
@@ -227,6 +227,7 @@ func (v *Vulcain) Apply(req *http.Request, rw http.ResponseWriter, responseBody
 	}
 	if addPreloadToVary {
 		responseHeaders.Add("Vary", "Preload")
+		rw.WriteHeader(http.StatusEarlyHints)
```

Comment:
I don't think that it's the good place to send the 103 response (hence the test failure).

This should be done here:
https://github.com/dunglas/vulcain/blob/f2d77e93a75ea803607bbc816c7eb1d534a2e7bc/vulcain.go#L256

and here: https://github.com/dunglas/vulcain/blob/f2d77e93a75ea803607bbc816c7eb1d534a2e7bc/vulcain.go#L288

Maybe should we refactor the code to prevent sending many 103 responses and send it only when all Preload headers have been computed?

Reply:
Also, I think that Server Push should now be opt-in (through a config option). By default, we should only use 103 responses as they are the new standard. This can be done in another PR of course.

Reply:
I moved the `WriteHeader(103)` call to the end of `apply()` +  manually gathered the links added by `push()`  because for some reason, they are not automatically included in the 103 response. With this approach, only one 103 response is sent with all Links included. 
The code feels a bit hackish but it works as expected so unless you see door for improvement, this is ready to go on my side.